File: MoodMate.ts
type MoodKey = 'happy' | 'sad' | 'stressed' | 'angry' | 'romantic' | 'neutral';

interface MoodInfo {
  quote: string;
  song: string;
  food: string;
}

const moodData: Record<MoodKey, MoodInfo> = {
  happy: {
    quote: "Happiness is not by chance, but by choice.",
    song: "Pharrell - Happy",
    food: "Ice Cream"
  },
  sad: {
    quote: "Tears come from the heart, not the brain.",
    song: "Adele - Someone Like You",
    food: "Dark Chocolate"
  },
  stressed: {
    quote: "Just breathe. You are strong.",
    song: "Lo-fi Chill Mix",
    food: "Herbal Tea"
  },
  angry: {
    quote: "Speak when you are calm, not when you are burning.",
    song: "Eminem - Lose Yourself",
    food: "Grilled Chicken"
  },
  romantic: {
    quote: "Love is composed of a single soul inhabiting two bodies.",
    song: "Ed Sheeran - Perfect",
    food: "Strawberries & Wine"
  },
  neutral: {
    quote: "Balance is the key to everything.",
    song: "Lo-fi Beats",
    food: "Fresh Salad"
  }
};

function logMood(mood: MoodKey): void {
  const timestamp = new Date().toISOString();
  const history = JSON.parse(localStorage.getItem('moodHistory') || '[]') as {mood: MoodKey, time: string}[];
  history.push({ mood, time: timestamp });
  localStorage.setItem('moodHistory', JSON.stringify(history));
  console.log(`Logged mood "${mood}" at ${timestamp}`);
}

function displayMoodInfo(mood: MoodKey): void {
  const info = moodData[mood];
  console.log(`\nQuote: "${info.quote}"`);
  console.log(`Suggested Song: ${info.song}`);
  console.log(`Suggested Food: ${info.food}`);
}

function promptMood(): void {
  console.log("Welcome to MoodMate - The Emotional Console");
  const moods = Object.keys(moodData) as MoodKey[];
  moods.forEach((mood, index) => {
    console.log(`${index + 1}. ${mood.charAt(0).toUpperCase() + mood.slice(1)}`);
  });

  const choiceStr = prompt(`Enter choice (1-${moods.length}):`);
  if (!choiceStr) {
    console.log("No input received. Exiting.");
    return;
  }

  const choice = Number(choiceStr);
  if (isNaN(choice) || choice < 1 || choice > moods.length) {
    console.log("Invalid choice. Exiting.");
    return;
  }

  const selectedMood = moods[choice - 1];
  displayMoodInfo(selectedMood);
  logMood(selectedMood);
  console.log("\nYour mood has been logged. Take care!");
}

promptMood();

File: MoodMate.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <ctime>
#include <map>
using namespace std;

struct MoodInfo {
    string quote;
    string song;
    string food;
};

map<string, MoodInfo> moodData = {
    {"happy", {"Happiness is not by chance, but by choice.",
               "Pharrell - Happy",
               "Ice Cream"}},
    {"sad", {"Tears come from the heart, not the brain.",
             "Adele - Someone Like You",
             "Dark Chocolate"}},
    {"stressed", {"Just breathe. You are strong.",
                  "Lo-fi Chill Mix",
                  "Herbal Tea"}},
    {"angry", {"Speak when you are calm, not when you are burning.",
               "Eminem - Lose Yourself",
               "Grilled Chicken"}},
    {"romantic", {"Love is composed of a single soul inhabiting two bodies.",
                  "Ed Sheeran - Perfect",
                  "Strawberries & Wine"}},
    {"neutral", {"Balance is the key to everything.",
                 "Lo-fi Beats",
                 "Fresh Salad"}}
};

string getCurrentTime() {
    time_t now = time(0);
    char buf[80];
    strftime(buf, sizeof(buf), "%Y-%m-%d %X", localtime(&now));
    return string(buf);
}

void saveMoodLog(const string& mood) {
    ofstream file("mood_log.txt", ios::app);
    if (file.is_open()) {
        file << getCurrentTime() << " - Mood: " << mood << endl;
        file.close();
    } else {
        cerr << "Error: Unable to open mood_log.txt for writing.\n";
    }
}

int main() {
    cout << "Welcome to MoodMate - The Emotional Console\n";
    cout << "Choose your current mood:\n";
    cout << "1. Happy\n2. Sad\n3. Stressed\n4. Angry\n5. Romantic\n6. Neutral\n";
    cout << "Enter choice (1-6): ";

    int choice;
    cin >> choice;

    string moodKey;
    switch (choice) {
        case 1: moodKey = "happy"; break;
        case 2: moodKey = "sad"; break;
        case 3: moodKey = "stressed"; break;
        case 4: moodKey = "angry"; break;
        case 5: moodKey = "romantic"; break;
        case 6: moodKey = "neutral"; break;
        default:
            cout << "Invalid choice. Exiting.\n";
            return 1;
    }

    MoodInfo info = moodData[moodKey];

    cout << "\nQuote: \"" << info.quote << "\"\n";
    cout << "Suggested Song: " << info.song << "\n";
    cout << "Suggested Food: " << info.food << "\n";

    saveMoodLog(moodKey);

    cout << "\nYour mood has been logged. Take care!\n";
    return 0;
}
